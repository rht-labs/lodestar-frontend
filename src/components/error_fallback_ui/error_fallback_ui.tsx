import React from 'react';
import {
  EmptyState,
  PageSection,
  EmptyStateIcon,
  Title,
  EmptyStateBody,
} from '@patternfly/react-core';
import { OutlinedSmileBeamIcon } from '@patternfly/react-icons';
import { useConfig } from '../../context/config_context/config_hook';

export const ErrorFallbackUI = () => {
  const { appConfig } = useConfig();
  return (
    <PageSection style={{ height: '100%' }}>
      <>
        <EmptyState>
          <EmptyStateIcon icon={OutlinedSmileBeamIcon} />
          <Title size="lg" headingLevel="h4">
            You've found an unexpected feature.
          </Title>
          <EmptyStateBody>
            <p>
              First, try refreshing your browser and navigating back to this
              page.
            </p>
            <p>
              If that doesn't work, please send an email to&nbsp;
              <a href={`mailto:${appConfig.supportEmailAddress}`}>
                {appConfig.supportEmailAddress}
              </a>
              .
            </p>
          </EmptyStateBody>
        </EmptyState>
      </>
    </PageSection>
  );
};
