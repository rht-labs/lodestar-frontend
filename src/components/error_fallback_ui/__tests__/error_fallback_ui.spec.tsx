import React from 'react';
import { render } from '@testing-library/react';
import { ErrorFallbackUI } from '../error_fallback_ui';

describe('Error fallback UI', () => {
  test('matches snapshot', () => {
    expect(render(<ErrorFallbackUI />)).toMatchSnapshot();
  });
});
