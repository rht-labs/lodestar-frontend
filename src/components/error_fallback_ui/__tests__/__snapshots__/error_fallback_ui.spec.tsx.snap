// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error fallback UI matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section"
        style="height: 100%;"
      >
        <div
          class="pf-c-empty-state"
        >
          <div
            class="pf-c-empty-state__content"
          >
            <svg
              aria-hidden="true"
              class="pf-c-empty-state__icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 496 512"
              width="1em"
            >
              <path
                d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm84-143.4c-20.8 25-51.5 39.4-84 39.4s-63.2-14.3-84-39.4c-8.5-10.2-23.6-11.5-33.8-3.1-10.2 8.5-11.5 23.6-3.1 33.8 30 36 74.1 56.6 120.9 56.6s90.9-20.6 120.9-56.6c8.5-10.2 7.1-25.3-3.1-33.8-10.2-8.4-25.3-7.1-33.8 3.1zM136.5 211c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4s-52.7 29.3-56 71.4c-.3 3.7 2.1 7.2 5.7 8.3 3.4 1.1 7.4-.5 9.3-3.7l9.5-17zM328 152c-23.8 0-52.7 29.3-56 71.4-.3 3.7 2.1 7.2 5.7 8.3 3.5 1.1 7.4-.5 9.3-3.7l9.5-17c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4z"
                transform=""
              />
            </svg>
            <h4
              class="pf-c-title pf-m-lg"
            >
              You've found an unexpected feature.
            </h4>
            <div
              class="pf-c-empty-state__body"
            >
              <p>
                First, try refreshing your browser and navigating back to this page.
              </p>
              <p>
                If that doesn't work, please send an email to 
                <a
                  href="#"
                />
                .
              </p>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section"
      style="height: 100%;"
    >
      <div
        class="pf-c-empty-state"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 496 512"
            width="1em"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm84-143.4c-20.8 25-51.5 39.4-84 39.4s-63.2-14.3-84-39.4c-8.5-10.2-23.6-11.5-33.8-3.1-10.2 8.5-11.5 23.6-3.1 33.8 30 36 74.1 56.6 120.9 56.6s90.9-20.6 120.9-56.6c8.5-10.2 7.1-25.3-3.1-33.8-10.2-8.4-25.3-7.1-33.8 3.1zM136.5 211c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4s-52.7 29.3-56 71.4c-.3 3.7 2.1 7.2 5.7 8.3 3.4 1.1 7.4-.5 9.3-3.7l9.5-17zM328 152c-23.8 0-52.7 29.3-56 71.4-.3 3.7 2.1 7.2 5.7 8.3 3.5 1.1 7.4-.5 9.3-3.7l9.5-17c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4z"
              transform=""
            />
          </svg>
          <h4
            class="pf-c-title pf-m-lg"
          >
            You've found an unexpected feature.
          </h4>
          <div
            class="pf-c-empty-state__body"
          >
            <p>
              First, try refreshing your browser and navigating back to this page.
            </p>
            <p>
              If that doesn't work, please send an email to 
              <a
                href="#"
              />
              .
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
